### ISSUE TITLE
Feature Request: extra key components

### ISSUE THREAD

**Issue by pySilver** [peterbe/django-cache-memoize#49]
Sometimes we know, that cache state should depend on some external data like `some dependency last update datetime`. It would be nice if `extra` args would be supported so one can pass smth like

```python
@cache_memoize(60, extra=('foo', obj.pk, foo.unix_timestamp))
def foo(*args, **kwargs):
    return 42
```

This also solves #35

**Comment by raiyankamal** [comment]
Could the `key_generator_callable` help in this case?

https://github.com/peterbe/django-cache-memoize/blob/ef449fee0d90daac87ca709fa2e3d00b150f7725/src/cache_memoize/__init__.py#L20

**Comment by pySilver** [comment]
@raiyankamal yes and no. It is quickly become tedious when you need this extra parameters in literally every usage of the decorator. Such as when you want your cache to be versioned.
